px.py

```diff
@@ -825,13 +909,30 @@ def get_destination(self):
                 dprint("%s => %s + %s" % (self.path, ipport, path))
 
                 if ipport[0] in State.noproxy:
+                    dprint("Direct connection from noproxy configuration")
                     self.path = path
                     return ipport
 
-        if not State.proxy_server:
-            return False
+        # Get proxy mode and servers straight from load_proxy to avoid threading issues
+        (proxy_mode, self.proxy_servers) = load_proxy()
+        if proxy_mode in [MODE_AUTO, MODE_PAC]:
+            proxy_str = find_proxy_for_url(
+                ("https://" if "://" not in self.path else "") + self.path)
+            if proxy_str == "DIRECT":
+                ipport = netloc.split(":")
+                ipport[1] = int(ipport[1])
+                dprint("Direct connection from PAC")
+                self.path = path
+                return tuple(ipport)
```

Comment:
Is this needed?

Reply:
I only added "self.path = path" and this is indeed necessary. For direct connections the protocol/host/port part has to be stripped. For direct connections from noproxy configuration this was already present in the old code (some lines above). But it's needed for direct connections from PAC too. Otherweise HTTP commands for DIRECT connections would contain the whole URL. It seems that most servers can handle this, but I found an Apache/Typo3 installation which doesn't resolve the requested page right in this case.